lib/githubbot.ts

```diff
@@ -0,0 +1,403 @@
+/*
+Copyright 2016 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import * as ProcBot from './procbot';
+import * as Promise from 'bluebird';
+import * as _ from 'lodash';
+import * as fs from 'fs';
+import * as jwt from 'jsonwebtoken';
+import { AlertLevel } from './procbot';
+
+// The GithubAPI Promise implies built-in ES6 Promises. Therefore when assigning
+// the Promise to use as Bluebird, VS Code reports an error. Need a good
+// fix for this that doesn't require type declarations?
+const GithubApi = require('github');
```

Comment:
I did some more investigation. The deep deep problem is that a Bluebird promise instance doesn't have the same type as a full real-world promise instance, because it doesn't have a `[Symbol.toStringTag]` key ([bluebird bug](https://github.com/petkaantonov/bluebird/issues/1277), [corresponding type definitions bug](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/11027#issuecomment-248642717)).

The Github types are determinedly asking for something with exactly the same type as a real implementation (`typeof Promise`), not just something that merely fulfills the spec, so the error's actually arguably right: bluebird isn't sufficient for what the github types are asking for. If, hypothetically, github did expect that symbol key, it'd break. Really, github should probably be asking for a `PromiseConstructorLike`, which is a way looser interface that just says you look like something that probably follows the spec.

It doesn't do that though, and it seems like there's other people looking at proper solutions on the Bluebird side, so for now I'd just ignore it. If you use proper types here, but use `Promise: <any> Promise` in the github call on line 204, I think everything else will keep typing happily, and you can skip merrily past the problem.